r2-core/src/main/java/com/linkedin/r2/message/QueryTunnelUtil.java

```diff
@@ -408,6 +421,12 @@ else if(existingQuery.isEmpty())
     }
     requestBuilder.setEntity(entity);
     requestBuilder.setHeaders(h);
+
+    String overrideHeader = request.getHeader(HEADER_METHOD_OVERRIDE);
+    if (!VALID_HTTP_VERBS.contains(overrideHeader)) {
+      LOG.warn("Invalid HTTP method override header " + overrideHeader + " was received, rejecting request.");
```

Comment:
Since overrideHeader can contain anything, let's not log the actual string here, unless there's a way to sanitize it.

Reply:
Done